cluster/kubernetes/release.go

```diff
@@ -49,16 +58,63 @@ func (c *Kubectl) connectArgs() []string {
 	return args
 }
 
-func (c *Kubectl) kubectlCommand(args ...string) *exec.Cmd {
-	cmd := exec.Command(c.exe, append(c.connectArgs(), args...)...)
-	cmd.Stdout = c.stdout
-	cmd.Stderr = c.stderr
-	return cmd
+func (c *Kubectl) execute(logger log.Logger, errs cluster.SyncError) {
+	defer c.changeSet.clear()
```

Comment:
This implicit statefulness makes me feel queasy -- it feels like you would really want to lock the applier while you staged things, then execute; or, you'd construct a set of staged actions, and supply that to execute. I guess it doesn't matter too much if the methods aren't exported.

Reply:
Yep you're totally right here. Why did I stash this data in the Kubectl struct again? 🤷‍♂️